### DISCUSSION TITLE
toggle button

### DISCUSSION THREAD

**Discussion by cruiser2** [arendst/Tasmota#22515] [General]
Just an opinion: 
I don‘t like your new default toggle button concept (observed in 14.3.0.7) even when I recognize the space saving (of the ON/OFF line). The relay status (ON or OFF) is now expressed in the color of the button of which one is the regular blue and the other a slightly dimmer color. Besides the fact that this color combination might be hard to read on a smartphone under certain daylight conditions they are generally not in line with the far more recognized colors for “go-no go” statuses: green and red. For an eye catching clarity I even would like to see the words ON or OFF again.

**Comment by arendst** [comment]
Thx for input.

If you like you can change the button colors with commands `WebColor11` (default) and `WebColor20` (Off).

The words ON and OFF are gone forever as in a 32 relay configuration it was a complete mess.

**Comment by joba-1** [comment]
Ok just updated a device to see it. Sorry, but I **HATE** it:

No more a clear message over the state of a central function of the device but color guessing. :/

Really, for an admin interface that I mainly use to checkout correct functionality and state representation, where it is so easy for a logic inversion to sneak in on all levels from hardware to smart home I will really miss that tasmota speaks a clear language to me! 

I'm sure you thought about it, but here is my kind appeal anyways: Please introduce some "if buttons < n" logic :)

**Comment by barbudor** [comment]
I just discover this discussion and I also gave it a try
I'll join the crusade as really doesn't like it, even if I can change the colours

I understand the concern for large number of relays but IMHO this is still an edge case.
How many Tasmota devices have more than 8 or even more than 4 relay/light/power ?

Please consider a `if` or at least a `#if` to restore the prvious view

Thanks

**Reply by Wintespe** [reply]
I typically switch one relay on/off in my Tasmota applications. I would like a clear, easily recognizable statement about this on/off state. Thanks.

**Comment by arendst** [comment]
For those who hate change, still use WorPerfect 4.2 and Windows 10, I'll try to add ON/OFF back for up to the first 8 power buttons...

**Comment by arendst** [comment]
How about this:

![image](https://github.com/user-attachments/assets/1578f749-faa2-40cc-90be-b999a189e178)

**Comment by joba-1** [comment]
love u 2 😍

**Comment by arendst** [comment]
Implemented in latest dev release. Use `SetOption161 1` to get rid of state text (I default show state text now to prevent heart attacks)

**Reply by cruiser2** [reply]
Thank you for taking care (even of my heart)!

**Reply by sfromis** [reply]
I guess that most users will not even be aware of `SetOption161` when they are not hating on the change :grin:

**Reply by Wintespe** [reply]
Thanks for the nice stuff  😁

**Reply by joba-1** [reply]
> I guess that most users will not even be aware of `SetOption161` when they are not hating on the change 😁

well, it once was about slider updates - now it has a much better use 🙂
And about hate: my psychotherapist said I have to let my feelings out - sorry about that  😉

**Reply by barbudor** [reply]
Thanks a lot @arendst

**Comment by scargill** [comment]
@arendst - thanks for the great work.... Ok, got the latest tasmota dev this morning - tried colours - I can't seem to separate the on off colors from the other buttons like config, information etc.  Could you please give an example of say - red for on, blue for off - leaving the remaining buttons (config, info etc) alone at a different colour?

**Reply by sfromis** [reply]
Specific colors depends on what theme you've selected, standard ones:
https://tasmota.github.io/docs/WebUI/#themes
The color for "off" is the last one, `WebColor20`, and the general setup is that the color for "off" is "less prominent" while the color for "on" is the same as the menu buttons below, but you can set up your own if you prefer.

You can see details in the build config overrides for [WebColor](https://github.com/arendst/Tasmota/blob/620fade73c902352574aa9675c7a438764c18f19/tasmota/my_user_config.h#L190-L237), but you do not need to do a build to change them; they are persistent if you use the `WebColor` command to change.

**Comment by raspitoaster** [comment]
Testing release 14.4.1, I also noticed the missing ON/OFF line - and I like the idea for Tasmota devices with more than one button.
But for a Tasmota device with just one button, the main webpage is hard to read.
What about implementing a COLOR_BUTTON_ON in addition to COLOR_BUTTON_OFF ?

**Reply by sfromis** [reply]
When realizing that the color for "off" stands out by being a different color (subdued), I'd not call it "hard to read". And note that the rudimentary web UI is not really intended for daily usage, but mainly for admin/test.

Still, the old separate ON/OFF indicators in the web UI are only suppressed if you have `SetOption161 1`

**Comment by PrasadNikapitiya** [comment]
How tasmota interface modifies the toggle add
